sdk/identity/azure-identity/README.md

```diff
@@ -305,6 +305,8 @@ variables:
 
 ### Username and password
 
+> **Warning**: Username and password authentication doesn't support multifactor authentication and is **deprecated**.
```

Comment:
This text should also be added at https://github.com/Azure/azure-sdk-for-python/blob/472e45cc3789bb7623e2eb317a59ef84c1ea88fc/sdk/identity/azure-identity/azure/identity/_credentials/environment.py#L48